hil/client/base.py

```diff
@@ -60,9 +62,51 @@ def check_response(self, response):
             except ValueError:  # No JSON request body; typical
                                 # For methods PUT, POST, DELETE
                 return
-        else:
+        try:
             e = json.loads(response.content)
             raise FailedAPICallException(
                 error_type=e['type'],
                 message=e['msg'],
             )
+        # Catching responses that do not return JSON
+        except ValueError:
+            return response.content
+
+
+def _find_reserved(string, slashes_ok=False):
+    """Returns a list of illegal characters in a string"""
+    if slashes_ok:
+        p = r"[^A-Za-z0-9 /$_.+!*'(),-]+"
+    else:
+        p = r"[^A-Za-z0-9 $_.+!*'(),-]+"
+    return list(x for l in re.findall(p, string) for x in l)
```

Comment:
well, except for this line. There's a lot going on here, would be nice if we split it to make it more readable. But I won't block the PR on this.

Reply:
I might be wrong, but I think this is one of those cases where the list comprehension is actually less complex than if I didn't use a one-liner.  I can add a comment for clarification if that helps.